drivers/gpio/gpio_stm32.c

```diff
@@ -225,6 +225,35 @@ static void gpio_stm32_set_exti_source(int port, int pin)
 #endif
 }
 
+static int gpio_stm32_get_exti_source(int pin)
+{
+	uint32_t line = gpio_stm32_pin_to_exti_line(pin);
+	int port;
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	port = LL_GPIO_AF_GetEXTISource(line);
+#elif CONFIG_SOC_SERIES_STM32MP1X
+	port = LL_EXTI_GetEXTISource(line);
+#elif defined(CONFIG_SOC_SERIES_STM32G0X)
```

Comment:
Then for coherency,  use `#if defined(CONFIG_SOC_SERIES_STM32F1X)` if applying @mnkp's comment